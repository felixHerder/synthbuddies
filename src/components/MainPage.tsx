import React from "react";
import Cardlist from './Cardlist';
import Searchbox from './Searchbox';
import Scroll from './Scroll';
import Header from './Header';
import ErroBoundry from './ErrorBoundry';
import "./MainPage.css";

export interface IRobot {
  name: string;
  id: number;
  email: string;
}
interface IAppProps {
  searchField: string;
  robots: Array<IRobot>;
  isPending: boolean;
  onRequestRobots(): void;
  onSearchChange(): void;
}
interface IAppState {
  robots: Array<IRobot>;
  searchField: string;

}

class MainPage extends React.Component<IAppProps, IAppState> {

  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    return this.props.robots.filter((elem) => elem.name.toLowerCase().includes(this.props.searchField.toLowerCase()));
  }

  render(): JSX.Element {
    return this.props.isPending ?
      <h1>Loading synths...</h1>
      :
      (<div className="tc">
        <Header />
        <Searchbox searchChange={this.props.onSearchChange} />
        <Scroll>
          <ErroBoundry>
            <Cardlist robots={this.filterRobots()} />
          </ErroBoundry>
        </Scroll>
      </div>);
  }
}

export default MainPage;