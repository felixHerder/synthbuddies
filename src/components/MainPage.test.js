import { shallow, mount, render } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps} />);
});

it('renders Mainpage without crashing', () => {
  expect(wrapper).toMatchSnapshot();
})


it('filters robots correctly 2', () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: "John",
      email: "john@gmail.com"
    }],
    searchField: 'a',
    isPending: false
  }
  const filteredRobots = []
  const wrapper3 = shallow(<MainPage {...mockProps3} />);
  expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
});

it('filters robots correctly 3', () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 3,
      name: "John",
      email: "john@gmail.com"
    }],
    searchField: 'a',
    isPending: true
  }
  const filteredRobots = []
  const wrapper3 = shallow(<MainPage {...mockProps3} />);
  expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
});