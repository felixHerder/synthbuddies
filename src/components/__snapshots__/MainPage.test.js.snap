// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Mainpage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <Searchbox />
  <Scroll>
    <ErrorBoundry>
      <Cardlist
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
